modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/dcrm/api/FAPIDCRValidationsTestCase.java

```diff
@@ -0,0 +1,183 @@
+                        "Signature validation failed for the software statement"
+                }
+        };
+    }
+
+    private void changeISConfiguration() throws Exception {
+
+        log.info("Adding entity id of SSOService to deployment.toml file");
+        String carbonHome = Utils.getResidentCarbonHome();
+        File defaultConfigFile = getDeploymentTomlFile(carbonHome);
+        File configuredIdentityXML = new File(getISResourceLocation() + File.separator + "oauth"
+                + File.separator + "dcr-fapi-validation-enabled.toml");
+        serverConfigurationManager = new ServerConfigurationManager(isServer);
+        serverConfigurationManager.applyConfigurationWithoutRestart(configuredIdentityXML,
+                defaultConfigFile, true);
+        serverConfigurationManager.restartGracefully();
+    }
+
+    @Test(alwaysRun = true, groups = "wso2.is", priority = 11,
+            description = "Check FAPI validations, PPID and SSA during DCR", dataProvider = "dcrConfigProvider")
+    public void validateErrorScenarios(JSONObject requestJSON, String errorCode, String errorMessage) throws Exception {
+
+        HttpPost request = new HttpPost(DCRUtils.getPath(tenant));
+        request.addHeader(HttpHeaders.AUTHORIZATION, DCRUtils.getAuthzHeader(username, password));
+        request.addHeader(HttpHeaders.CONTENT_TYPE, OAuthDCRMConstants.CONTENT_TYPE);
+        StringEntity entity = new StringEntity(requestJSON.toJSONString());
+        request.setEntity(entity);
+        HttpResponse response = client.execute(request);
+
+        assertEquals(response.getStatusLine().getStatusCode(), 400, "Service Provider " +
+                "has not been created successfully");
+        JSONObject errorResponse = DCRUtils.getPayload(response);
+        assertEquals(errorResponse.get("error"), errorCode);
+        assertEquals(errorResponse.get("error_description"), errorMessage);
+    }
+    @Test(alwaysRun = true, groups = "wso2.is", priority = 11,
+            description = "Check FAPI validations, PPID and SSA during DCR", dataProvider = "dcrConfigProvider")
+    public void validateErrorScenariosForDCRUpdate(JSONObject requestJSON, String errorCode, String errorMessage)
+            throws Exception {
+
+        // Create application.
+        HttpPost request = new HttpPost(DCRUtils.getPath(tenant));
+        JSONObject registerRequestJSON = DCRUtils.getRegisterRequestJSON("request6.json");
+
+        request.addHeader(HttpHeaders.AUTHORIZATION, DCRUtils.getAuthzHeader(username, password));
+        request.addHeader(HttpHeaders.CONTENT_TYPE, OAuthDCRMConstants.CONTENT_TYPE);
+        StringEntity entity = new StringEntity(registerRequestJSON.toJSONString());
+        request.setEntity(entity);
+        ObjectMapper mapper = new ObjectMapper();
+
+        HttpResponse response = client.execute(request);
+        assertEquals(response.getStatusLine().getStatusCode(), 201, "Service Provider " +
```

Comment:
The created application should be deleted.

Reply:
Also the test is for error scenarios. why do we need to check if a creation is successful?

Reply:
this is to check whether the validations work during the application update flow